packages/server/src/fhir/operations/set-accounts.ts

```diff
@@ -141,7 +141,10 @@ export async function setResourceAccounts(
     accounts: accounts,
     account: accounts?.[0],
   };
-  await repo.updateResource(target);
+  if (!repo.canPerformInteraction(AccessPolicyInteraction.UPDATE, target)) {
+    throw new OperationOutcomeError(forbidden);
+  }
+  await systemRepo.updateResource(target);
```

Comment:
Right now `updateResource` eventually calls `getAccounts` which has differing behavior for extended mode which I believe is intentional? Paging @mattwiller